lxd/device/disk.go

```diff
@@ -106,6 +108,10 @@ func (d *disk) validateConfig() error {
 		return fmt.Errorf("The recursive option is only supported for additional bind-mounted paths")
 	}
 
+	if d.config["source"] == "ceph" && (d.config["ceph.cluster_name"] == "" || d.config["ceph.user_name"] == "") {
+		return fmt.Errorf("Ceph mapping is missing cluster name or user name property")
```

Comment:
So that's not correct. We should error if either of those options are set with a source which isn't "ceph" or "cephfs".

But the options aren't required by themselves and should have the same defaults as we do in storage pools, so:
 - ceph.user_name defaults to admin
 - ceph.cluster_name defaults to ceph

Reply:
Where would we set the defaults? Or do we always check if it is empty when they are used?

Reply:
I just put them where they're used